core/src/test/java/org/testcontainers/junit/wait/strategy/HttpWaitStrategyTest.java

```diff
@@ -30,6 +34,26 @@ public void testWaitUntilReadyWithSuccess() {
         waitUntilReadyAndSucceed(createShellCommand("200 OK", GOOD_RESPONSE_BODY));
     }
 
+    /**
+     * Ensures that HTTP requests made with the HttpWaitStrategy can be enriched with user defined headers,
+     * although the test web server does not depend on the header to response with a 200, by checking the
+     * logs we can ensure the HTTP request was correctly sent.
+     */
+    @Test
+    public void testWaitUntilReadyWithSuccessWithCustomHeaders() {
+        HashMap<String, String> headers = new HashMap<>();
+        headers.put("baz", "boo");
+        String logs = waitUntilReadyAndSucceedWithLogs(
+            startContainerWithCommand(createShellCommand("200 OK", GOOD_RESPONSE_BODY),
+                createHttpWaitStrategy(ready)
+                    .withHeader("foo", "bar")
+                    .withHeaders(headers)
+            ));
+
+        assertThat(logs, containsString("foo: bar"));
```

Comment:
I would much prefer a black box test with a server requiring the headers.
On the other hand, if you want logs, you can just access the container, which is returned by `startContainerWithCommand`.

Reply:
I got a bit blocked by how the tests works, the netcat utility is quite limited, I was afraid that I was going to have to touch too many things in order to achieve the result needed!
If you think that I should follow this path, what kind of service should I spin off? A mock service, custom bash script? what'd be appropriate in your opinion?

Reply:
After thinking a bit more about it, I think your approach is actually valid and does not involve too much overhead.

I rewrote the method like this:

```java
    @Test
    public void testWaitUntilReadyWithSuccessWithCustomHeaders() {
        HashMap<String, String> headers = new HashMap<>();
        headers.put("baz", "boo");
        GenericContainer container = startContainerWithCommand(createShellCommand("200 OK", GOOD_RESPONSE_BODY),
            createHttpWaitStrategy(ready)
                .withHeader("foo", "bar")
                .withHeaders(headers)
        );
        waitUntilReadyAndSucceed(container);

        String logs = container.getLogs();

        assertThat(logs, containsString("foo: bar"));
        assertThat(logs, containsString("baz: boo"));
    }
```

Reply:
Thanks, I've updated it with your suggestion!